test/ModuleInterface/discard_interface.swift

```diff
@@ -0,0 +1,28 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-emit-module-interface(%t/Library.swiftinterface) %s -module-name Library -verify
+// RUN: %target-swift-typecheck-module-from-interface(%t/Library.swiftinterface) -I %t
+// RUN: %FileCheck %s < %t/Library.swiftinterface
+
+// This test makes sure that discard and _forget are emitted correctly in the
+// interfaces. We expect that if you use the old name _forget, you'll get that
+// in the interface file.
+
+// CHECK: @_alwaysEmitIntoClient public consuming func AEIC_discard() { discard self }
+// CHECK: @inlinable public consuming func inlinable_discard() { discard self }
+
+// CHECK: @_alwaysEmitIntoClient public consuming func AEIC_forget() { _forget self }
+// CHECK: @inlinable public consuming func inlinable_forget() { _forget self }
```

Comment:
It's good that this happens, but I'm not quite sure why it works. `PrintAST::visitDiscardStmt` specifically emits `discard`, so this must somehow be getting carried-over from the original parsed source file??